### ISSUE TITLE
[Bug] x:DataType prevents binding in ListView.GroupHeaderTemplate

### ISSUE THREAD

**Issue by MaxFmi** [xamarin/Xamarin.Forms#13481]
### Description
We're using ListView.GroupHeaderTemplate with a custom layout. The issue arises when we set x:DataType to ListView or on one of its parent xaml tags like ContentPage.

### Steps to Reproduce

1. Creat a ContentPage with
2. Add a ListView with GroupHeaderTemplate 
3. Use a ViewModel as BindingContext
4. Set x:DataType to ContentPage or ListView or both  
5. Add Binding to your items in your GroupHeaderTemplate // could be as trivial like a Label
6. Start your application and look at your bound items in GroupHeaderTemplate

### Expected Behavior
Views in GroupHeaderTemplate should show the bound content. 

### Actual Behavior
Views in GroupHeaderTemplate aren't showing the bound content. 

### Basic Information

- Version with issue: Xamarin.Forms 4.8.0.1269
- Last known good version: -
- Platform Target Frameworks: 
  - iOS:  13.7

### Environment
Rider 2020.2.4 on a Mac

### Workaround
Not using x:DataType

**Comment by PureWeen** [comment]
@MaxFmi can you attach a reproduction please?

Does the same issue happen if you are using CollectionView?

**Comment by salgiza** [comment]
Just in case this is related, because it might seem so reading between lines in the description: if the x:DataType does not exactly correspond to the class that is being binded, bindings do not work.

As a workaround for #14191, I said: "ok, I'll set one of the many classes that implement the interface as the x:DataType", that way at least I'll get Intellisense to work. And it worked for Intellisense... but when running the app, bindings only worked when the ContentView was showing that particular class, but not when showing any of the other classes that implemented the same interface.

TLDR; even if the binded class has properties with the names specified in the XAML, they won't be binded if they class is not the same that has been specified in the x:DataType.
